agent/acs/session/payload_responder.go

```diff
@@ -106,6 +107,16 @@ func (pmHandler *payloadMessageHandler) addPayloadTasks(payload *ecsacs.PayloadM
 			allTasksOK = false
 			continue
 		}
+
+		// Note: If we receive an EBS-backed task, we'll also received an incomplete volume configuration in the list of Volumes
+		// To accomodate this, we'll first check if the task IS EBS-backed then we'll mark the corresponding Volume object to be
+		// of type "attachment". This volume object will be replaced by the newly created EBS volume configuration when we parse
+		// through the task attachments.
+		volName, ok := hasEBSAttachment(task)
+		if ok {
+			initializeAttachmentTypeVolume(task, volName)
+		}
```

Comment:
Is this tested in any unit tests? The test update seems to be in the input and not in assertions?

Reply:
+1

Reply:
Not directly, I was hoping to do this in a follow up. A lot of the testing was mostly done through manual testing.